From 5f7b9b045d4713e8ff27a4fc8b2799669c1b823a Mon Sep 17 00:00:00 2001
From: Carlo Lobrano <c.lobrano@gmail.com>
Date: Tue, 20 May 2025 09:34:03 +0200
Subject: [PATCH] podman-etcd: fix listen-peer-urls binding (#2049)

This change ensures learner etcd listens on all interfaces for peer
connections, resolving accessibility issues.

Fix: OCPBUGS-56447
---
 heartbeat/podman-etcd | 12 +++---------
 1 file changed, 3 insertions(+), 9 deletions(-)

diff --git a/heartbeat/podman-etcd b/heartbeat/podman-etcd
index 3a2323260..6762112ec 100755
--- a/heartbeat/podman-etcd
+++ b/heartbeat/podman-etcd
@@ -436,15 +436,9 @@ prepare_env() {
 	ETCD_PEER_CERT=$(get_env_from_manifest "ETCDCTL_CERT")
 	ETCD_PEER_KEY=$(get_env_from_manifest "ETCDCTL_KEY")
 
-	if is_learner; then
-		LISTEN_CLIENT_URLS="$NODEIP"
-		LISTEN_PEER_URLS="$NODEIP"
-		LISTEN_METRICS_URLS="$NODEIP"
-	else
-		LISTEN_CLIENT_URLS="0.0.0.0"
-		LISTEN_PEER_URLS="0.0.0.0"
-		LISTEN_METRICS_URLS="0.0.0.0"
-	fi
+	LISTEN_CLIENT_URLS="0.0.0.0"
+	LISTEN_PEER_URLS="0.0.0.0"
+	LISTEN_METRICS_URLS="0.0.0.0"
 }
 
 archive_data_folder()
